### PR TITLE
Reduce unnecessary S3 read flow after restart

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/7513

Problem Summary:
The segment delta index on wn is useless now. But it introduces high CPU and high S3 read-write flow after restart. And now wn could serve much more data than the coupled arch.
We should disable the segment place index under disagg arch.

### What is changed and how it works?

Only generate place index tasks after restart when deployed with coupled arch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
image
image
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp [hunk 1]
```diff
@@ -275,11 +275,16 @@ void DeltaMergeStore::setUpBackgroundTask(const DMContextPtr & dm_context)
 
     blockable_background_pool_handle = blockable_background_pool.addTask([this] { return handleBackgroundTask(true); });
 
-    // Generate place delta index tasks
-    for (auto & [end, segment] : segments)
+    // Under disagg mode, a write node could serve large amount of data, place delta index tasks
+    // after restart is useless and waste of S3 reading. Only do it when deployed non-disagg mode.
+    if (!global_context.getSharedContextDisagg()->notDisaggregatedMode())
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:280]:
```suggestion
    if (global_context.getSharedContextDisagg()->notDisaggregatedMode())
```

[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp]: You're right, it is a mistake. Fixed

**Review by lidezhu — APPROVED** [review]

**Pull request merged**